### PR TITLE
Reader Comments: Show separator for last leaf comments

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #17476

This PR implements separator logic for the comment thread. Additionally, the separator is always hidden for the last row (as per the design).

I couldn't find a better term for this 😅 , but a "last leaf" comment is the last descendant of a top-level comment. I've also added a note in the code, but here's an example:

```
- comment 1
  - comment 2
    - comment 3 --> this is the last leaf of comment 1.
- comment 4
  - comment 5
    - comment 6
  - comment 7 --> this is the last leaf of comment 4.
- comment 8
```

## To Test

- Enable the `newCommentThread` feature flag.
- Go to Reader > any comment thread.
- 🔍  Verify that the cell separators are correctly shown for every "last leaf" comment.
- Rotate the device to landscape.
- 🔍 Verify that the cell separators are displayed correctly.

## Regression Notes
1. Potential unintended areas of impact
n/a. Feature is hidden behind a flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a. Feature is hidden behind a flag.

3. What automated tests I added (or what prevented me from doing so)
n/a. Feature is hidden behind a flag.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.swift [hunk 1]
```diff
@@ -38,8 +38,8 @@ import UIKit
     func handleHeaderTapped() {
         guard let post = post,
               allowsPushingPostDetails else {
-            return
-        }
+                  return
+              }
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.swift:42]: Classic Xcode 13 formatting. 🙄

**Comment by ScoutHarris** [comment]
I noticed one visual oddity. On an iPad (in this case iPad Pro 9.7-inch iOS15):
- In portrait, view post comments.
- Rotate to landscape.
- Notice the separator "artifacts" on the left.

landscape

**Review by ScoutHarris — APPROVED** [review]
I noted one UI issue. But since this is a WIP I'll go ahead and approve to not hold you up.

**Comment by dvdchr** [comment]
> I noted one UI issue. 

Thank you! This should be fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17549/commits/31b7f3f58c15c43bdc730f79a60f07c96cc85d2e. I noticed that this happened only after rotating from portrait to landscape. It doesn't happen when opening the comment thread while already in landscape orientation 😅. Here's a screenshot after rotating from portrait to landscape:

Simulator Screen Shot - iPad Air (4th generation) - 2021-11-25 at 16 02 20

I've also tested to ensure that this does not cause any issues in iPhone. 🙂

**Pull request merged**